### PR TITLE
Adding provider ipfinder

### PR DESCRIPTION
**Pull request by benemohamed**
Hi
Adding provider ipfinder.

### CODE REVIEW

**Comment by jbelien** [comment]
Hello,

Can you make sure your PR passes all the tests ?
Thanks a lot !

File: phpunit.xml.dist [hunk 1]
```diff
@@ -37,7 +37,7 @@
         <server name="PICKPOINT_API_KEY" value="YOUR_API_KEY" />
         <server name="TOMTOM_MAP_KEY" value="YOUR_MAP_KEY" />
         <server name="GEOCODE_EARTH_API_KEY" value="YOUR_GEOCODE_EARTH_API_KEY" />
-        <server name="IPFINDER_API_KEY" value="YOUR_API_KEY" />
+        <server name="IPFINDER_API_KEY" value="ent" />
```
Comment (jbelien) [inline phpunit.xml.dist]: Why this has been changed ?
It should be `YOUR_API_KEY`.

**Comment by jbelien** [comment]
Could you also add some specific tests ?
Check another [IP provider](https://github.com/geocoder-php/Geocoder#ip) to have "examples".

File: phpunit.xml.dist (hunk 1 above)
Comment (benemohamed) [reply phpunit.xml.dist]: Thanks I forgot

File: src/Provider/IpFinder/Tests/IpFinderTests.php [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    }
+
+    /**
+     * @expectedException \Geocoder\Exception\InvalidArgument
+     * @expectedExceptionMessage No API Key was specified, invalid API Key.
+     */
+    public function testGeocodeWith401Code()
+    {
+        $json = <<<'JSON'
+{
```
[the remark this answers is not in the document]
Comment (atymic) [reply src/Provider/IpFinder/Tests/IpFinderTests.php]: (and below)

File: src/Provider/IpFinder/Tests/IpFinderTests.php [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+         "code": 405
+      }
+   ]
+}
+JSON;
+        $provider = new IpFinder($this->getMockedHttpClient($json), 'asdsad');
+        $result = $provider->geocodeQuery(GeocodeQuery::create('1.0.0.0'));
+    }
+
+    public function testGeocodeWithIPv4()
```
Comment (atymic) [inline src/Provider/IpFinder/Tests/IpFinderTests.php]: Can we have a few more example/tets?

Comment (atymic) [reply src/Provider/IpFinder/Tests/IpFinderTests.php]: It seems the state is returned in the response but not set in this provider. Can you add that?

File: src/Provider/IpFinder/composer.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
+{
+    "name": "geocoder-php/ip-ipfinder-provider",
```
Comment (atymic) [inline src/Provider/IpFinder/composer.json]: Why `ip-ipfinder-provider` not `ipfinder-provider`

**Comment by atymic** [comment]
I'm a bit concerned that you made this PR without disclosing you affiliation with the provider (it seems you created it).

I can't even find the website searching for the company name "ipfinder"

File: src/Provider/IpFinder/CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# Change Log
+
+The change log describes what is "Added", "Removed", "Changed" or "Fixed" between each release.
+
+## 1.0.0
```
[the remark this answers is not in the document]
Comment (jbelien) [reply src/Provider/IpFinder/CHANGELOG.md]: No, no, 1.0.0 is good :)

**Comment by benemohamed** [comment]
@atymic 
the website https://ipfinder.io/

**Comment by jbelien** [comment]
@benemohamed Would you have time to finish this PR (and merge our `master` to fix the conflict) ? Thanks.

**Comment by Chris53897** [comment]
Looks like the website https://ipfinder.io/ is down.

**Comment by jbelien** [comment]
Thanks for the notice @Chris53897, I guess we can close this PR then.

**Pull request closed without merging**